src/app.js

```diff
@@ -0,0 +1,275 @@
+    let assetDurationAttr = this.toTimecodeStr(this.#mediaInfo.frameCount)
+    let clipStartAttr = this.toTimecodeStr(startFrame)
+    let clipDuration = endFrame - startFrame
+    let clipDurationAttr = this.toTimecodeStr(clipDuration)
+    let clipOffsetAttr = this.toTimecodeStr(this.#currentOffset)
+    if (this.#mediaInfo.hasVideo) {
+      asset = `<asset id="${assetIDAttr}" format="r1" name="${assetNameAttr}" start="0/1s" duration="${assetDurationAttr}" hasVideo="1" hasAudio="1" audioSources="1"><media-rep kind="original-media" src="${assetNameAttr}"/></asset>`
+      clip = `<asset-clip ref="${assetIDAttr}" format="r1" tcFormat="NDF" name="${assetNameAttr}" start="${clipStartAttr}" duration="${clipDurationAttr}" offset="${clipOffsetAttr}" enabled="1"><adjust-transform scale="1 1" anchor="0 0" position="0 0"/></asset-clip>`
+    } else {
+      asset = `<asset id="${assetIDAttr}" name="${assetNameAttr}" start="0/1s" duration="${assetDurationAttr}" hasAudio="1" audioSources="1"><media-rep kind="original-media" src="${assetNameAttr}"/></asset>`
+      clip = `<asset-clip ref="${assetIDAttr}" name="${assetNameAttr}" start="${clipStartAttr}" duration="${clipDurationAttr}" offset="${clipOffsetAttr}" lane="2" enabled="1"/>`
+    }
+    this.#assetList.push(asset)
+    this.#clipList.push(clip)
+    this.#currentAssetID++
+    this.#currentOffset += clipDuration
+  }
+  stringify () {
+    let sequenceTCStartAttr = this.toTimecodeStr(this.#initialOffset)
+    let sequenceDurationAttr = this.toTimecodeStr(
+      this.#currentOffset - this.#initialOffset
+    )
+    if (this.#mediaInfo.hasVideo) {
+      this.#sequenceStr = `<sequence format="r0" tcFormat="NDF" tcStart="${sequenceTCStartAttr}" duration="${sequenceDurationAttr}"><spine>`
+    } else {
+      this.#sequenceStr = `<sequence format="r0" tcFormat="NDF" tcStart="${sequenceTCStartAttr}" duration="${sequenceDurationAttr}"><spine><gap name="Gap" start="${sequenceTCStartAttr}" duration="${sequenceDurationAttr}" offset="${sequenceTCStartAttr}">`
+    }
+    return (
+      this.#startStr +
+      this.#assetList.join('') +
+      this.#eventStr +
+      this.#sequenceStr +
+      this.#clipList.join('') +
+      this.#endStr
+    )
+  }
+} // class FCPXML
+
+let ffmpeg
+
+// Check browser support and load libraries
+async function initialize () {
+  if (typeof SharedArrayBuffer === 'undefined') {
+    document.getElementById('message').innerHTML =
+      'Error: Please use latest Chrome/Firefox/Edge'
+  }
+  ffmpeg = FFmpeg.createFFmpeg({ log: true })
+  await ffmpeg.load()
+}
+
+function download (url, fileName) {
+  let link = document.createElement('a')
+  link.href = url
+  link.download = fileName
+  document.body.appendChild(link)
+  link.click()
+  link.remove()
+}
+
+// Convert editList to a proper output format
+function generateOutput (mediaInfo, editList, format) {
+  let output
+  if (format === 'fcpxml') {
+    output = new FCPXML(mediaInfo)
+  } // add formats here
```

Comment:
will leave this part to Aidan. There might be a better way than using if else to handle multiple formats

Reply:
Well, the format will eventually be user-determined, an if...else block or a switch statement to map each format is sufficient and is very easy to change. Don't have to add complexity if we there's no need to